### ISSUE TITLE
Problem starting the module mod_bosh for host <<"localhost">> 

### ISSUE THREAD

**Issue by sandeepjangir** [processone/ejabberd#2839]
Hi all,

This is my ejabberd detail:
ejabberd version: 16
System version: Erlang/OTP 17
OS: CentOS
installed by: compiling by source code

**I have set up the bosh or mod_bosh like this way:**

```
  - 
    port: 5280
    ip: "::"
    certfile: "/usr/local/etc/ssl.pem"
    module: ejabberd_http
    request_handlers:
#      "/websocket": ejabberd_http_ws
      "/bosh": mod_bosh
    ##  "/pub/archive": mod_http_fileserver
    web_admin: true
    http_bind: true
    tls: true
    starttls: true
```

**the module is like this way**

```
modules: 
  mod_bosh: {}
```

**but my ejabberd is failed to start and getting these error logs:**

```
=erl_crash_dump:0.3
Thu Mar 21 08:45:22 2019
Slogan: Problem starting the module mod_bosh for host <<"localhost">> 
 options: []
 error: undef
[{mod_bosh,start,[<<"localhost">>,[]],[]},
 {gen_mod,start_module,3,[{file,"src/gen_mod.erl"},{line,98}]},
 {
System version: Erlang/OTP 17 [erts-6.1] [source] [64-bit] [smp:16:2] [async-threads:10] [kernel-poll:true]
Compiled: Thu Mar  7 12:14:45 2019
```

can anyone help me what is wrong with this configuration or why I am getting this annoying error?

Thanks in advance

**Comment by licaon-kter** [comment]
That's too old, not sure if still supported, can't you update first?

**Comment by badlop** [comment]
mod_bosh was first included in ejabberd 16.12: https://github.com/processone/ejabberd/commit/ebadcf71c2e12e379afd31829c58717565d541c4 . You didn't specify your ejabberd version, but it's probably older, that module is not included, and the error message says "undef", which means undefined.

**Comment by sandeepjangir** [comment]
upgrading the ejabberd version to the latest, resolved the issue thanks for the answers;
